### PR TITLE
feat: added account managent resource, docs, examples and test cases

### PR DESCRIPTION
**Pull request by RavitejaSurampudi**
Added new resource for the account management API's which contains create and update a sub account .
Added test cases  on the  update and import of the resource.
Added documentation on how to use this resource

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

### CODE REVIEW

File: newrelic/resource_newrelic_account_management.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	}
+
+	return resourceNewRelicAccountRead(ctx, d, meta)
+}
+
+func resourceNewRelicAccountDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	diags = append(diags, diag.Diagnostic{
+		Severity: diag.Warning,
+		Summary:  "Account cannot be deleted via Terraform. Please contact support",
```
Comment (NSSPKrishna) [inline newrelic/resource_newrelic_account_management.go:122]: Can we add a link to the support docs here?

File: newrelic/resource_newrelic_account_management_test.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccNewRelicAccountManagement(t *testing.T) {
+	//t.Skip("Skipping the test case,as there is no delete api to delete the account")
```
Comment (NSSPKrishna) [inline newrelic/resource_newrelic_account_management_test.go:17]: Remove it if not required

File: newrelic/resource_newrelic_account_management.go (hunk 1 above)
Comment (RavitejaSurampudi) [reply newrelic/resource_newrelic_account_management.go]: done

File: newrelic/resource_newrelic_account_management_test.go (hunk 2 above)
Comment (RavitejaSurampudi) [reply newrelic/resource_newrelic_account_management_test.go]: done

File: website/docs/r/account_management.hmtl.markdown [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "newrelic"
+page_title: "New Relic: newrelic_account_management"
+sidebar_current: "docs-newrelic-resource-account_management"
```
[the remark this answers is not in the document]
Comment (RavitejaSurampudi) [reply website/docs/r/account_management.hmtl.markdown]: done

**Pull request merged**